xdsl/rewriter.py

```diff
@@ -39,7 +39,7 @@ def replace_op(op: Operation,
         if not isinstance(new_ops, list):
             new_ops = [new_ops]
         if new_results is None:
-            new_results = [] if len(new_ops) == 0 else new_ops[-1].results
+            new_results = cast(List[SSAValue | None], new_ops[-1].results)
```

Comment:
What happens here when `new_ops` is empty? This was handled before but not anymore.

Reply:
Reverted, let's discuss!